### ISSUE TITLE
[TW-976] Can't Add Task with Tag?

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1011]
_Michelle Crane on 2010-10-18T09:55:59Z says:_ 

 Just started using the 1.9.3 beta.  Wanted to see if I could use something like (at)waiting as a tag.  Fail.  So, tried to use +waiting.  Noticed that I can add +waiting after the task has been completed, but I can't add it when I create the task. See trace:

<pre>
F:\Dropbox\My Dropbox\TextFiles>t add testing
Created task 69.
F:\Dropbox\My Dropbox\TextFiles>t 69 +waiting
Modified 1 task.
F:\Dropbox\My Dropbox\TextFiles>t add testing +waiting
Cannot add task because the uuid '7631d451-db8d-ad54-1e1f-955179f5c4f3' is not unique.
F:\Dropbox\My Dropbox\TextFiles>t add fjdaklf;djask; +waiting
Cannot add task because the uuid '7631d451-db8d-ad54-1e1f-955179f5c4f3' is not unique.
F:\Dropbox\My Dropbox\TextFiles>t add bob +hello
Cannot add task because the uuid 'aeed64e5-ad25-a8b8-8368-fa46f9439b13' is not unique.
F:\Dropbox\My Dropbox\TextFiles>t add +hello bob
Cannot add task because the uuid 'd19c796a-afd7-1d18-ba1a-bc52b2d88795' is not unique.
F:\Dropbox\My Dropbox\TextFiles>
</pre>

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2010-10-18T09:55:59Z
Modified: 2014-02-09T02:13:25Z
```

**Comment by taskwarrior** [comment]
_Federico Hernandez on 2010-10-18T17:35:59Z says:_ 

 Thanks for your report Michelle. So far we can't reproduce your observation on our main development platforms:

<pre>
[fredde(at)supai ~]$ t add testing
Created task 1.
[fredde(at)supai ~]$ t 1 +waiting
Modified 1 task.
[fredde(at)supai ~]$ t add testing +waiting
Created task 2.
[fredde(at)supai ~]$ t list

ID Project Pri Due Active Age     Description                                  
2                         15 secs testing                                      
1                         41 secs testing                                      

2 tasks
[fredde(at)supai ~]$ t add fjdaklf;djask; +waiting
Created task 3.
* bash: djask: command not found
* bash: +waiting: command not found
[fredde(at)supai ~]$ t add bob +hello
Created task 4.
[ultrafredde(at)supai ~]$ t add +hello bob
Created task 5.
</pre>

Regarding the error when doing a @t add fjdaklf;djask; +waiting@ please note that @;@ is a special character to the shell and needs to be escaped - even when being part of a string. The same applies to @'@.

We will have to look into the differences of bash under cmd.exe in cygwin. Are you running bash with cmd.exe? What happens if you use an alternative terminal like rxvt?

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2010-10-18T21:46:26Z says:_ 

 I cannot reproduce the bug, but I do get the same results as Fredde, exactly.

Michelle, is there any chance that your "t" command points to something other than 1.9.3?

**Comment by taskwarrior** [comment]
_Michelle Crane on 2010-10-19T07:43:47Z says:_ 

 I can't seem to add anything with 1.9.3.  Here is another test, with the task.exe spelled out:

<pre>
F:\Dropbox\My Dropbox\TextFiles>"F:\Dropbox\My Dropbox\TextFiles\task-1.9.3.beta1\src\task.exe" add testing
Cannot add task because the uuid '7631d451-db8d-ad54-1e1f-955179f5c4f3' is not unique.

F:\Dropbox\My Dropbox\TextFiles>"F:\Dropbox\My Dropbox\TextFiles\task-1.9.3.beta1\src\task.exe" add haven't used this before
Cannot add task because the uuid '300b92eb-8ef0-326c-8b79-0958127bd470' is not unique.

F:\Dropbox\My Dropbox\TextFiles>"F:\Dropbox\My Dropbox\TextFiles\task-1.9.3.beta1\src\task.exe" ls testing
ID Project Pri Countdown * + R Description
* - ------- --- --------- - - - -----------------------------------------------------------------------------
61 55866          22 hrs *     Continue testing the 2nd button to make sure I like all of the error messages
1 task

F:\Dropbox\My Dropbox\TextFiles>"F:\Dropbox\My Dropbox\TextFiles\task-1.9.3.beta1\src\task.exe" add ppppppppppppp
Cannot add task because the uuid '74f8fced-653b-4fd6-7485-a17ec67c6a8d' is not unique.
</pre>

**Comment by taskwarrior** [comment]
_Federico Hernandez on 2010-10-19T09:25:43Z says:_ 

 Michelle,

Let's fall back to the basics. Can you run:

$ task version
$ task show
$ task color
$ task list

Let's see if these non-adding commands work for you

**Comment by taskwarrior** [comment]
_Michelle Crane on 2010-10-19T09:37:27Z says:_ 

 Okay, I've got no problems with anything but add yet.  Here's the first bit of the trace.  (Basically, I'm using a task.bat (1.9.3) and a t.bat (1.9.2) in conjunction.

<pre>
F:\Dropbox\My Dropbox\TextFiles>type task.bat
(at)echo off
"F:\Dropbox\My Dropbox\TextFiles\task-1.9.3.beta1\src\task.exe" %*

F:\Dropbox\My Dropbox\TextFiles>task version
task 1.9.3.beta1 built for cygwin-ncurses
Copyright (C) 2006 - 2010 P. Beckingham, F. Hernandez.
Taskwarrior may be copied only under the terms of the GNU General Public License, which may be found in the taskwarrior source
kit.
Documentation for taskwarrior can be found using 'man task', 'man taskrc', 'man task-tutorial', 'man task-color', 'man task-faq'
or at http://taskwarrior.org

F:\Dropbox\My Dropbox\TextFiles>task show
Config variable                 Value
_forcecolor                     no
active.indicator                *
alias.export                    export.yaml
alias.export.vcalendar          export.ical
alias.getpri                    list priority.any:
</pre>

Skipping...color works, list works.  And, if I add a task with 1.9.2, I can then do all sorts of things with it with 1.9.3:

<pre>
F:\Dropbox\My Dropbox\TextFiles>task add testing
Cannot add task because the uuid '2c0e8dee-7385-25cc-a7ad-06660735d840' is not unique.

F:\Dropbox\My Dropbox\TextFiles>t add testing
Created task 71

F:\Dropbox\My Dropbox\TextFiles>task 71 pri:L
Modified 1 task.

F:\Dropbox\My Dropbox\TextFiles>task 71 due:today
Modified 1 task.

F:\Dropbox\My Dropbox\TextFiles>task 71 app "appending"
Appended 'appending' to task 71.
Appended 1 task.

F:\Dropbox\My Dropbox\TextFiles>task 71 pre "prepending"
Prepended 'prepending' to task 71.
Prepended 1 task.

F:\Dropbox\My Dropbox\TextFiles>task 71 ann "annotating"
Annotated 71 with 'annotating'.

F:\Dropbox\My Dropbox\TextFiles>task info 71
Name        Value
* ---------- ------------------------------------
ID          71
Description prepending testing appending
            [2010-10-19 09:34] annotating
Status      Pending
Priority    L
Due         2010-10-19.00:00

F:\Dropbox\My Dropbox\TextFiles>task 71 /testing/replacing/
Modified 1 task.

F:\Dropbox\My Dropbox\TextFiles>t info 71
Name        Value
* ---------- ------------------------------------
ID          71
Description prepending replacing appending
            [2010-10-19 09:34] annotating
Status      Pending
Priority    L
Due         2010-10-19.00:00
UUID        b0de4f39-d294-4d30-9b2b-cb86a13d5cc0
Entered     2010-10-19.09:33 (3 mins)
</pre>

**Comment by taskwarrior** [comment]
_Federico Hernandez on 2010-10-19T10:40:50Z says:_ 

 THX Michelle. Let's concentrate now on the adding a task issue and the problem with UUIDs.

1) Run "task add foo". Capture the uuid it complains about, grep for that uuid in ~/.task/*.data and let us know if you find it.

2) Run "grep HAVE_SRANDOM task-1.9.3beta1/auto.h"

3) What version of cygwin are you using? Is it fully patched?

**Comment by taskwarrior** [comment]
_Michelle Crane on 2010-10-19T11:02:59Z says:_ 

 Responses below:

1) <pre>
F:\Dropbox\My Dropbox\TextFiles>task add foo
Cannot add task because the uuid '1c543532-d28e-3479-aa8b-55344b0ec47e' is not unique.
</pre>

Ugh.  Ran a findstr and found the following counts of that UUID:

<pre>
F:\Dropbox\My Dropbox\TextFiles\.task>find /c "1c543532-d28e-3479-aa8b-55344b0ec47e" *.data
* --------- COMPLETED.DATA: 9
* --------- PENDING.DATA: 3
* --------- UNDO.DATA: 14
</pre>

I opened pending.data, and yup, three different tasks have that identical UUID!  Here are the lines, with the sekrit guts missing:

<pre>
Line 11: [annotation_1280435819:"xxx" annotation_1283265379:"xxx" description:"xxx" entry:"1275305400" project:"10611" status:"pending" tags:"pt" uuid:"1c543532-d28e-3479-aa8b-55344b0ec47e"]
	Line 15: [description:"xxx" entry:"1279757071" status:"pending" tags:"someday,career" uuid:"1c543532-d28e-3479-aa8b-55344b0ec47e"]
	Line 36: [description:"xxx" entry:"1284993466" project:"55866.cleanup" status:"pending" uuid:"1c543532-d28e-3479-aa8b-55344b0ec47e"]
</pre>

For sure the first two were created ages ago, and probably the third one was created pre-1.9.3 as well.

2) Don't know what this is doing, but here the result:

<pre>
F:\Dropbox\My Dropbox\TextFiles>grep HAVE_SRANDOM task-1.9.3.beta1\auto.h
1. define HAVE_SRANDOM 1
</pre>

3) I honestly have no idea how to tell what version of cygwin I'm running.  I updated it a few months ago (when I was looking for an editor to use with the task edit command.  I basically only touch it when I need to add something.

Hope this helps!

**Comment by taskwarrior** [comment]
_Federico Hernandez on 2010-10-19T14:53:06Z says:_ 

 Thanks again, Michelle.

1) It seems that your 1.9.3.beta1 creates identical uuids compared to 1.9.1 and 1.9.2. We will now have to see if this is cygwin related or not and for which environment. That is why we wanted to know about 2) the random generator. As for 3) we might have to ask you later on to update depending on our findings.

**Comment by taskwarrior** [comment]
_Cory Donnelly on 2010-10-22T10:12:05Z says:_ 

 Hi Michelle, here's how to check your version under most *nix variants, it should work for cygwin as well:

<pre>
$ uname -srv
Darwin 8.11.1 Darwin Kernel Version 8.11.1: Wed Oct 10 18:23:28 PDT 2007; root:xnu-792.25.20~1/RELEASE_I386</pre>

**Comment by taskwarrior** [comment]
_Cory Donnelly on 2010-10-22T10:21:48Z says:_ 

 Michelle, from your trace it looks like you're using taskwarrior from the Windows commandline, cmd.exe, and not from the bash shell included with cygwin. Is that the case?

**Comment by taskwarrior** [comment]
_Michelle Crane on 2010-10-22T10:31:29Z says:_ 

 (at)Cory,

1) Yes, I do access taskwarrior from the Windows side.

2) on the Cygwin side, the version (thanks for the command) is
CYGWIN_NT-5.1 1.7.5(0.225/5/3) 2010-04-12 19:07.

Hope that helps.

Just out of curiosity, I tried adding a task, using 1.9.3.beta1 from the cygwin side. It didn't seem to have a problem, the first time.  When I tried it a second time, it was a problem.  See:

<pre>
xxx /cygdrive/f/dropbox/mydrop~1/textfiles
$ task add testing
Created task 71.

xxx /cygdrive/f/dropbox/mydrop~1/textfiles
$ task ver

task 1.9.3.beta1 built for cygwin-ncurses
Copyright (C) 2006 - 2010 P. Beckingham, F. Hernandez.

Taskwarrior may be copied only under the terms of the GNU General Public License
, which may be found in the taskwarrior source
kit.

Documentation for taskwarrior can be found using 'man task', 'man taskrc', 'man
task-tutorial', 'man task-color', 'man task-faq'
or at http://taskwarrior.org


xxx /cygdrive/f/dropbox/mydrop~1/textfiles
$ task add testing4
Cannot add task because the uuid 'afda6a5b-02a1-244a-d0ac-6a474f434e9f' is not u
nique.
</pre>

The UUID in the failed task addition matches the UUID on two older tasks.'

<pre>
Line 38: [description:"xxx" entry:"1285355084" project:"xxx" status:"pending" uuid:"afda6a5b-02a1-244a-d0ac-6a474f434e9f"]
Line 57: [annotation_1287490828:"xxx" description:"xxx" entry:"1287490789" priority:"L" project:"xxx" status:"pending" uuid:"afda6a5b-02a1-244a-d0ac-6a474f434e9f"]
</pre>
